src/utils/hooks/useModal.ts

```diff
@@ -0,0 +1,35 @@
+import { useState } from "react";
+
+export type Modal = {
+  [key: string]: boolean;
+};
+export type OpenModal = (modalName: string) => void;
+export type CloseModal = (modalName: string) => void;
+
+export type UseModal = {
+  modals: Modal;
+  openModal: OpenModal;
+  closeModal: CloseModal;
+}
+
+const useModal = () => {
+  const [modals, setModals] = useState<Modal>({});
+
+  const openModal = (modalName: string) => {
+    setModals((prevModals) => ({
+      ...prevModals,
+      [modalName]: true,
+    }));
+  };
+
+  const closeModal: CloseModal = (modalName: string) => {
```

Comment:
타입을 안잡아줘도 괜찮을 것 같아요~
```suggestion
  const closeModal = (modalName: string) => {
```